Sources/SQLKit/Query/SQLColumnConstraintAlgorithm.swift

```diff
@@ -0,0 +1,133 @@
+/// Column constraint algorithms used by `SQLConstraint`
+public enum SQLColumnConstraintAlgorithm: SQLExpression {
+    /// `PRIMARY KEY`column constraint.
+    case primaryKey(autoIncrement: Bool)
+
+    /// `NOT NULL` column constraint.
+    case notNull
+
+    /// `UNIQUE` column constraint.
+    case unique
+
+    /// `CHECK` column constraint.
+    case check(SQLExpression)
+
+    /// `COLLATE` column constraint.
+    case collateConstraint(name: SQLExpression)
+
+    /// `DEFAULT` column constraint.
+    case defaultConstraint(SQLExpression)
```

Comment:
Can we use backticks here to make this `case default` ?

Reply:
This is a side-effect of the fact that a "`default`" case with one associated value conflicts with all `static` "`default`" functions of one argument. We can either have 
```
case `default`(SQLExpression)
``` 
*or*
```
case defaultConstraint(SQLExpression)
static func `default`(_ value: String)
static func `default`<T: BinaryInteger>(_ value: T)
static func `default`<T: FloatingPoint>(_ value: T)
static func `default`(_ value: Bool)
static func `default`(_ value: SQLExpression)
```

The latter provides a lot of utility (including being able to still write `.default(SQLRaw("something"))`) and the only cost is that when you are actually pattern-matching on a `SQLColumnConstraintAlgorithm` value you end up using `.defaultConstraint`.

The aforementioned (and numerous) convenience constructors come into play when adding a column to a `SQLColumnBuilder` via the `column(_: dataType: _: _:)` method.

Reply:
An alternative spelling of the case that also would not collide is 
```swift
case `default`(expression: SQLExpression)
```
and I guess I don't have a preference.